src/main/js/html.js

```diff
@@ -133,9 +148,24 @@
             bpd: null, /* block progression direction (lr, rl, tb) */
             ruby: null, /* is ruby present in a <p> */
             textEmphasis: null, /* is textEmphasis present in a <p> */
-            rubyReserve: null /* is rubyReserve applicable to a <p> */
+            rubyReserve: null, /* is rubyReserve applicable to a <p> */
+            options: options || {}
```

Comment:
9116902 still appears to mutate the existing options. I think this should do it:
```suggestion
            options: {}
        };

        if (options) { 
            if (options.colorAdjust)
            context.options.colorAdjust = preprocessColorMapOptions(options.colorAdjust);
            
            var bgcColorElements = ['region', 'body', 'div', 'p', 'span'];
            var propName;
            for (var bgcei in bgcColorElements)
            {
                propName = bgcColorElements[bgcei] + backgroundColorAdjustSuffix;
                if (options[propName])
                context.options[propName] = preprocessColorMapOptions(options[propName]);
            }
        }
```

Reply:
Sigh. That suggestion would trash all the other options settings that aren't colour adjustments, so that wouldn't work either. We need to work on a copy that isn't a reference. Or I could make the keys for the preprocessed colour maps a different value to the passed-in ones instead of the same value?

Reply:
bdd8530 should do it, but I don't like it!

Reply:
I think I've got a compromise. This will copy the initial *colorAdjust references but then overwrite them when `preprocessColorMapOptions` happens. It also removes the need to copy across the other properties added in bdd8530.
```suggestion
            options: Object.assign({}, options) || {},
        };
```

Reply:
I thought of that too @jlks but it breaks on build I think because the grunt script is set to refuse ES6-only features, which includes `Object.assign` :-(

Reply:
Oh! I don't get any `grunt:build` errors. I see `Object.entries()` being used inside `preprocessColorMapOptions` does that need to change too? I see it gets added as a polyfill in the minified distribution just the same as using `Object.assign`.

Reply:
Huh, okay, maybe I didn't actually _try_ it, but just assumed it would fail. I'll make this fix (tomorrow) - it will definitely clean things up.

Reply:
OK done in 13dc9a5